www/src/pages/table/OntSearchComponent.tsx

```diff
@@ -41,59 +46,132 @@ function useSearchComponent(
   props: SearchProps
 ) {
   const store = useStores()[storeName];
-  const seqInvoker = new SequentialFuncInvoker();
+  const [searchOptions, setSearchOptions] = useState<SearchOptions[]>();
 
-  // when the provided value is not in the store, fetch it
-  useEffect(() => {
-    if (props.value === undefined) {
-      return;
-    }
+  const loaderOption: SearchOptions = {
+    id: "",
+    label: <Spin style={{ width: "100%", marginTop: "2px" }} size="large" />,
+    filterlabel: ``,
+    value: ``,
+  };
 
-    if (Array.isArray(props.value)) {
-      // leverage the fact that the three stores are not re-fetched
-      if (!store.refetch) {
-        store.fetchByIds(props.value);
-      } else {
-        store.fetchByIds(props.value.filter((id) => !store.records.has(id)));
-      }
-    } else if (store.get(props.value) === undefined) {
-      store.fetchById(props.value);
+  // search for additional values if it's not in the list
+  const onSearch = (query: string) => {
+    if (query === "") {
+      return;
     }
-  }, [props.value]);
+    const searchResults: SearchOptions[] = [];
+    loaderOption.filterlabel = query;
 
-  // gather all options already in the store, leverage the fact
-  // that property store is readonly
-  const options = useMemo(() => {
-    const options = [];
-    for (const r of store.iter()) {
-      options.push({
-        value: r.id,
-        label: r.readableLabel,
-      });
+    setSearchOptions([loaderOption]);
+    if (storeName === "classStore") {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: ClassTextSearchResult) => {
+            searchResults.push({
+              type: "class",
+              id: searchResult.id,
+              label: (
+                <LabelComponent
+                  id={searchResult.id}
+                  description={searchResult.description}
+                  label={searchResult.label}
+                />
+              ),
+              filterlabel: `${searchResult.label} (${searchResult.id})`,
+              value: `${searchResult.id}`,
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
```

Comment:
no need to use spreading. you should search and remove all redundant spreading operator in the code

Reply:
btw you can simply the code, no need for an extra then function.

Reply:
fixed this